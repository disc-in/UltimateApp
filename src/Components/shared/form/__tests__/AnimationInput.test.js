import React from 'react';
import { render } from '@testing-library/react-native';
import { Formik } from 'formik';
import { NavigationContext } from '@react-navigation/native';

import AnimationInput from '../AnimationInput';

describe('AnimationInput', () => {
  // fake NavigationContext value data
  const navContext = {
    isFocused: () => true,
    // addListener returns an unscubscribe function.
    addListener: jest.fn(() => jest.fn()),
  };

  it('renders correctly', () => {
    const { toJSON } = render(
      <NavigationContext.Provider value={navContext}>
        <Formik initialValues={{ title: 'My title' }}>
          <AnimationInput fieldName="title" label="Title" />
        </Formik>
      </NavigationContext.Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
