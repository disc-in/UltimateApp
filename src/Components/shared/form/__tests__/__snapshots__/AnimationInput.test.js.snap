// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimationInput renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 10,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 5,
      }
    }
  >
    <Text>
      Title
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#fff",
          "borderColor": "#2a9e91",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "center",
          "marginRight": 10,
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 0,
          "width": "auto",
        }
      }
      testID="editAnimation"
    >
      <Text
        style={
          Object {
            "color": "#2a9e91",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Edit
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#fff",
          "borderColor": "#2a9e91",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "center",
          "marginRight": 10,
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 0,
          "width": "auto",
        }
      }
      testID="deleteAnimation"
    >
      <Text
        style={
          Object {
            "color": "#2a9e91",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Remove
      </Text>
    </View>
  </View>
</View>
`;
